### ISSUE TITLE
Sonarr Wont start

### ISSUE THREAD

**Issue by mitchy112** [Sonarr/Sonarr#2431]
## Bug Report
Since i restarted my ubuntu machine today, sonarr wont start.
So what did I do I created a new server and installed everything again, but sill no result. (followed the github tutorial)
I think its one of the latest updates which broke it, since i did a massive update install on the old machine.
If i start sonarr manually it just says:
config$ sudo mono /opt/NzbDrone/NzbDrone.exe

Press enter to exit...

### System Information/Logs

**Sonarr Version:**
Laterst
**Operating System:**
Ubuntu 16.04

**.net Framework (Windows) or mono (macOS/Linux) Version:**
Mono JIT compiler version 5.10.0.140 (tarball Sat Feb 24 15:33:47 UTC 2018)

**Link to Log Files (debug or trace):**
it doesnt make log files...
**Browser (for UI bugs):**
Doesnt even startup

**Comment by mechanimal82** [comment]
Same for me. Just updated Mono and it's broken all packages which rely on it from starting. Sonarr Logs don;t show anything... because it can't start. 

```
$ mono --version
Mono JIT compiler version 5.10.0.140 (tarball Sat Feb 24 15:33:47 UTC 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug
        Interpreter:   yes
        LLVM:          supported, not enabled.
        GC:            sgen (concurrent by default)
```
Possibly linked to this issue; but I'm not technical enough to be able to confirm:

https://github.com/mono/mono/issues/7285

**Comment by mechanimal82** [comment]
Looks like it's not just us: https://www.reddit.com/r/sonarr/comments/80g1e2/broken_after_upgrade_to_mono_510_stable/

**Comment by icenine9** [comment]
Pin your version of mono to 5.8.0.127 for the time being. 

I did something like this (on Ubuntu):

apt remove mono-complete mono-devel (or whatever you installed for mono, I did mono-complete)
apt autoremove

I edited /etc/apt/sources.list.d/mono-official.list to read: 

deb http://download.mono-project.com/repo/ubuntu xenial/snapshots/5.8.0.127 main

Note in the above line, I locked to the 5.8.0.127 version, which still works with sonarr/radarr/jackett.

Then I did an apt update. 

Then I did an apt install mono-complete, that got me back down to 5.8.0.127. 

Hope this helps someone.

**Comment by icenine9** [comment]
Oh yeah - save your mono-official.list to another file before editing so you can put it back once this issue gets fixed.

**Comment by SybilSystem** [comment]
Great work around, thanks @icenine9

**Comment by campbc** [comment]
I am having issues with broken packages when trying to roll back to 5.8.0.127. I can reinstall to 5.10 just fine, but rolling back does not work for me without a lot of work it appears. Looks like I have to wait for a fix or reinstall my OS (would be easier) with the older version of Mono. 

Any ideas on what on Mono broke Sonarr and what kind of timeline we are looking at? Just wondering so I can manually download weekly shows.

**Comment by dutch2005** [comment]
http://www.mono-project.com/download/vs/#download-lin

The latest VS Mono release is: 5.8.0 Stable (5.8.0.108) e.g. it wont have the issue eighter

At first i tought it was an error on my side (as I am running 18.04 LTS alpha2), seems its a Mono issue ^^

**Comment by Taloth** [comment]
Based on the error I'm seeing, I really doubt it's linked to mono issue 7285.

**Comment by Taloth** [comment]
Yup, Definitely absolutely not a bug in mono. It's a bug in NLog that got fixed in their current preview 4.5 releases (currently at RC6).
I'll see if rc6 is stable enough to use.

**Comment by jonare77** [comment]
Strange, tried to roll back too version 5.8.0.127 with sucess, but still same problem with sonarr

**Comment by Taloth** [comment]
@jonare77 you might want to check the mono version again and the output of sonarr.

**Comment by animosity22** [comment]
I rolled back and no issue.

**Comment by aassaly** [comment]
For CentOS 6/7...
First:
`yum history list mono`
Then:
Look at the latest ID number in the table
Finally:
yum history undo <ID#>

**Comment by pcspecialist** [comment]
do you think the VS Mono release would be secure enough?  It would sure reduce the number of updates.

**Comment by dutch2005** [comment]
placing the file below in the sonarr directory (e.g. /opt/Nzbdrone (and renaming the old file) will make sonarr work on 5.8 - 5.10 and 5.13 (nightly)

https://download.sonarr.tv/hotfixes/regression-mono-5.10/NLog.dll.gz

**Comment by pcspecialist** [comment]
The new NLog.dll did the trick.  I replaced the old one with it

after which I unpinned mono (I previously followed icenine9's instructions ^^^^) by:
mv -f /etc/apt/sources.list.d/mono-official.list.original /etc/apt/sources.list.d/mono-official.list

Then updated to the latest stable mono by:
apt update
apt upgrade

**Comment by markus101** [comment]
Fixed in 2.0.0.5183.

Guide to manually upgrade is here: https://www.reddit.com/r/sonarr/comments/80ylur/new_sonarr_version_v2005183_and_mono_510_guide_if/

**Comment by s0m3f00l** [comment]
Thanks sirs!
